src/main.rs

```diff
@@ -0,0 +1,72 @@
+use std::env;
+use std::process;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Instant;
+use std::sync::mpsc::channel;
+
+const MICROS_PER_SEC: u64 = 1_000_000;
```

Comment:
if you convert to using time:: for timing, you'd probably define the number of ns per unit instead - eg:

```
const SECOND: u64 = 1_000_000_000;
const MICROSECOND: u64 = 1_000;
```